roachtest: sqlsmith/setup=empty/setting=no-ddl failed

[(roachtest).sqlsmith/setup=empty/setting=no-ddl failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2114210&tab=buildLog) on [master@bfa6307c292ef4dfed4a53cb99f506e6dab26533](https://github.com/cockroachdb/cockroach/commits/bfa6307c292ef4dfed4a53cb99f506e6dab26533):

```
																SELECT
																	jsonb_object_agg(cte_ref_412.col_9010::STRING, cte_ref_412.col_9010)::JSONB
																		AS col_9011
																FROM
																	with_1467 AS cte_ref_411, with_1467 AS cte_ref_412
																WHERE
																	true
																GROUP BY
																	cte_ref_412.col_9010
																HAVING
																	st_coveredby('0101000000000000000000F03F000000000000F03F':::GEOMETRY::GEOMETRY, '0101000000000000000000F03F000000000000F03F':::GEOMETRY::GEOMETRY)::BOOL
																LIMIT
																	1:::INT8
															)
														)
												)
													AS tab_5370 (col_9012) ON true,
											(
												VALUES
													(e'5\x12\x0bbCqY\x1bz':::STRING),
													(from_ip('\xed':::BYTES::BYTES)::STRING),
													(e'z\x0fh/\x05G*\x1e':::STRING),
													(e'\x0e\x11zG?Ox?':::STRING),
													(NULL),
													(e'Kl[\x06LxIL':::STRING)
											)
												AS tab_5371 (col_9013),
											(VALUES (':1tx)':::STRING), ('4AZ':::STRING)) AS tab_5372 (col_9014)
										LIMIT
											1:::INT8
									),
									3681474330:::OID
								),
								(NULL, NULL, 3400800720:::OID, 0:::OID),
								(NULL, e'*5~\x05':::STRING:::NAME, 0:::OID, 3493895375:::OID),
								(e'L\x05\x19D':::STRING, e'\x05f\nTx':::STRING:::NAME, NULL, 3508151978:::OID)
						)
							AS tab_5373 (col_9016, col_9017, col_9018, col_9019)
				)
		SELECT
			2213922715:::OID AS col_9020,
			cte_ref_413.col_9017 AS col_9021,
			cte_ref_413.col_9019 AS col_9022,
			cte_ref_413.col_9016 AS col_9023
		FROM
			with_1468 AS cte_ref_413
		WHERE
			NULL
		ORDER BY
			cte_ref_413.col_9019 DESC, cte_ref_413.col_9017 DESC, cte_ref_413.col_9018 DESC;

```

<details><summary>More</summary><p>


Artifacts: [/sqlsmith/setup=empty/setting=no-ddl](https://teamcity.cockroachdb.com/viewLog.html?buildId=2114210&tab=artifacts#/sqlsmith/setup=empty/setting=no-ddl)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Asqlsmith%2Fsetup%3Dempty%2Fsetting%3Dno-ddl.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

Looks like an opt issue cc @RaduBerinde:
```
06:20:01 test.go:325: test failure: 	sqlsmith.go:172,sqlsmith.go:202,test_runner.go:757: error: pq: internal error: unhandled op jsonb-object-agg
		stmt:
		WITH
			with_1466 (col_9005)
				AS (
					SELECT
						'0101000000000000000000F03F000000000000F03F':::GEOMETRY AS col_9005
					FROM
						(
							VALUES
								(2009821718:::OID, 764005869:::INT8, ARRAY[2319679505:::OID]),
								(1257313103:::OID, 393251616:::INT8, ARRAY[1110391882:::OID,2427102496:::OID,2273140837:::OID]),
								(
									1837310019:::OID,
									1424085777:::INT8,
									CASE
									WHEN true THEN ARRAY[2098323852:::OID,2350069855:::OID,1569434009:::OID]
									ELSE ARRAY[2676039481:::OID,517103106:::OID,2535779805:::OID,1928814912:::OID]
									END
								),
								(NULL, (-955316854):::INT8, NULL),
								(
									442885417:::OID,
									NULL,
									ARRAY[3647748847:::OID,0:::OID,2819954074:::OID,4193062288:::OID,2943014815:::OID]
								),
								(1384505778:::OID, 1381053916:::INT8, ARRAY[795530834:::OID])
						)
							AS tab_5365 (col_9002, col_9003, col_9004)
				),
			with_1468 (col_9016, col_9017, col_9018, col_9019)
				AS (
					SELECT
						*
					FROM
						(
							VALUES
								(
									'?,a:':::STRING,
									'':::STRING,
									COALESCE(CASE WHEN true THEN 3695586231:::OID ELSE 3639477942:::OID END, 2058471711:::OID),
									816126183:::OID
								),
								(
									e'\x1c|j':::STRING,
									NULL,
									(
										SELECT
											3448956394:::OID AS col_9015
										FROM
											(VALUES (NULL), ('14:13:00.944153-10:25:00':::TIMETZ)) AS tab_5368 (col_9009)
											LEFT JOIN (
													VALUES
														('1':::JSONB),
														(NULL),
														(
															'[{}, [[["KgIn-", []]], 0.10079074959009726], {}, [{}], true, {"T^6oRj": null}]':::JSONB
														),
														((NULL::JSONB->(-3065050135164090978):::INT8::INT8)::JSONB),
														(
															(
																WITH
																	with_1467 (col_9010)
																		AS (
																			SELECT
																				*
																			FROM
																				(VALUES (e'fK7q*\x1f^\n':::STRING), (NULL))
																					AS tab_5369 (col_9010)
																		)
																SELECT
																	jsonb_object_agg(cte_ref_412.col_9010::STRING, cte_ref_412.col_9010)::JSONB
																		AS col_9011
																FROM
																	with_1467 AS cte_ref_411, with_1467 AS cte_ref_412
																WHERE
																	true
																GROUP BY
																	cte_ref_412.col_9010
																HAVING
																	st_coveredby('0101000000000000000000F03F000000000000F03F':::GEOMETRY::GEOMETRY, '0101000000000000000000F03F000000000000F03F':::GEOMETRY::GEOMETRY)::BOOL
																LIMIT
																	1:::INT8
															)
														)
												)
													AS tab_5370 (col_9012) ON true,
											(
												VALUES
													(e'5\x12\x0bbCqY\x1bz':::STRING),
													(from_ip('\xed':::BYTES::BYTES)::STRING),
													(e'z\x0fh/\x05G*\x1e':::STRING),
													(e'\x0e\x11zG?Ox?':::STRING),
													(NULL),
													(e'Kl[\x06LxIL':::STRING)
											)
												AS tab_5371 (col_9013),
											(VALUES (':1tx)':::STRING), ('4AZ':::STRING)) AS tab_5372 (col_9014)
										LIMIT
											1:::INT8
									),
									3681474330:::OID
								),
								(NULL, NULL, 3400800720:::OID, 0:::OID),
								(NULL, e'*5~\x05':::STRING:::NAME, 0:::OID, 3493895375:::OID),
								(e'L\x05\x19D':::STRING, e'\x05f\nTx':::STRING:::NAME, NULL, 3508151978:::OID)
						)
							AS tab_5373 (col_9016, col_9017, col_9018, col_9019)
				)
		SELECT
			2213922715:::OID AS col_9020,
			cte_ref_413.col_9017 AS col_9021,
			cte_ref_413.col_9019 AS col_9022,
			cte_ref_413.col_9016 AS col_9023
		FROM
			with_1468 AS cte_ref_413
		WHERE
			NULL
		ORDER BY
			cte_ref_413.col_9019 DESC, cte_ref_413.col_9017 DESC, cte_ref_413.col_9018 DESC;
```

--- a/pkg/sql/opt/operator.go
+++ b/pkg/sql/opt/operator.go
@@ -382,9 +382,9 @@ func AggregatesCanMerge(inner, outer Operator) bool {
 		// while CountOp and CountRowsOp both output int values.
 		return outer == SumIntOp
 
-	case ArrayAggOp, AvgOp, ConcatAggOp, CorrOp, JsonAggOp,
-		JsonbAggOp, PercentileContOp, PercentileDiscOp, SqrDiffOp,
-		StdDevOp, StringAggOp, VarianceOp, StdDevPopOp, VarPopOp:
+	case ArrayAggOp, AvgOp, ConcatAggOp, CorrOp, JsonAggOp, JsonbAggOp,
+		JsonObjectAggOp, JsonbObjectAggOp, PercentileContOp, PercentileDiscOp,
+		SqrDiffOp, StdDevOp, StringAggOp, VarianceOp, StdDevPopOp, VarPopOp:
 		return false
 
 	default:
@@ -402,7 +402,8 @@ func AggregateIgnoresDuplicates(op Operator) bool {
 
 	case ArrayAggOp, AvgOp, ConcatAggOp, CountOp, CorrOp, CountRowsOp, SumIntOp,
 		SumOp, SqrDiffOp, VarianceOp, StdDevOp, XorAggOp, JsonAggOp, JsonbAggOp,
-		StringAggOp, PercentileDiscOp, PercentileContOp, StdDevPopOp, VarPopOp:
+		StringAggOp, PercentileDiscOp, PercentileContOp, StdDevPopOp, VarPopOp,
+		JsonObjectAggOp, JsonbObjectAggOp:
 		return false
 
 	default:

--- a/pkg/sql/opt/operator_test.go
+++ b/pkg/sql/opt/operator_test.go
@@ -1,0 +1,58 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package opt
+
+import (
+	"reflect"
+	"runtime"
+	"testing"
+)
+
+// TestAggregateProperties verifies that the various helper functions for
+// various properties of aggregations handle all aggregation operators.
+func TestAggregateProperties(t *testing.T) {
+	check := func(fn func()) bool {
+		ok := true
+		func() {
+			defer func() {
+				if x := recover(); x != nil {
+					ok = false
+				}
+			}()
+			fn()
+		}()
+		return ok
+	}
+
+	for _, op := range AggregateOperators {
+		funcs := []func(Operator) bool{
+			AggregateIgnoresDuplicates,
+			AggregateIgnoresNulls,
+			AggregateIsNeverNull,
+			AggregateIsNeverNullOnNonNullInput,
+			AggregateIsNullOnEmpty,
+		}
+
+		for _, fn := range funcs {
+			if !check(func() { fn(op) }) {
+				fnName := runtime.FuncForPC(reflect.ValueOf(fn).Pointer()).Name()
+				t.Errorf("%s not handled by %s", op, fnName)
+			}
+		}
+
+		for _, op2 := range AggregateOperators {
+			if !check(func() { AggregatesCanMerge(op, op2) }) {
+				t.Errorf("%s,%s not handled by AggregatesCanMerge", op, op2)
+				break
+			}
+		}
+	}
+}

